pilot/cmd/pilot-agent/status/server.go

```diff
@@ -177,14 +177,28 @@ func (s *Server) handleReadyProbe(w http.ResponseWriter, _ *http.Request) {
 	s.mutex.Unlock()
 }
 
+func isRequestFromLocalhost(r *http.Request) bool {
+	ip, _, err := net.SplitHostPort(r.RemoteAddr)
```

Comment:
I don't think this is secure enough, there are ways to modify this ip address i think.

Reply:
do you know how this is possible? I looked a bit and tried some things out and was not able to modify it

Reply:
I just searched it, and seems it is hard. And this has to be done in server side. So i think this is viable.

Reply:
If we use a TCP connection, it is not possible to fake the sender's address, because the connection is being negotiated. There is no problem with spoofing the sender's address for UDP traffic.